$OpenBSD: patch-plugins_tools_basictools_kis_tool_colorpicker_cc,v 1.1.1.1 2017/06/01 16:36:52 rsadowski Exp $

XXX Compiler threading support is not turned on in our boost

Index: plugins/tools/basictools/kis_tool_colorpicker.cc
--- plugins/tools/basictools/kis_tool_colorpicker.cc.orig
+++ plugins/tools/basictools/kis_tool_colorpicker.cc
@@ -21,8 +21,6 @@
 #include "kis_tool_colorpicker.h"
 #include <string.h>
 
-#include <boost/thread/locks.hpp>
-
 #include <QPoint>
 #include <QLayout>
 #include <QCheckBox>
@@ -57,7 +55,10 @@
 #include "kis_wrapped_rect.h"
 #include "kis_tool_utils.h"
 
+#include <thread>
+#include <mutex>
 
+
 namespace
 {
 // The location of the sample all visible layers in the combobox
@@ -114,7 +115,7 @@ void KisToolColorPicker::pickColor(const QPointF& pos)
     }
 
 
-    QScopedPointer<boost::lock_guard<KisImage> > imageLocker;
+    QScopedPointer<std::lock_guard<KisImage> > imageLocker;
 
     KisPaintDeviceSP dev;
 
@@ -123,7 +124,7 @@ void KisToolColorPicker::pickColor(const QPointF& pos)
         dev = currentNode()->projection();
     }
     else {
-        imageLocker.reset(new boost::lock_guard<KisImage>(*currentImage()));
+        imageLocker.reset(new std::lock_guard<KisImage>(*currentImage()));
         dev = currentImage()->projection();
     }
 
